filer/models/filemodels.py

```diff
@@ -506,16 +511,43 @@ def __lt__(self, other):
 
     @property
     def actual_name(self):
+        clean_name = self.clean_actual_name
+        if not self.sha1:
+            try:
+                self.generate_sha1()
+            except Exception:
```

Comment:
Set specific error type + ValueError found from tests.

In that case, at that later time, `self.actual_name` is not used to retrieve or make the URL, `self.file.name` is the used property. 
The self.file attributes will be up to date with the file and when the File is changed in a path/content affecting way, the file will be moved from `self.file.name` to `self.actual_name`.
This is why existing filer resources will still work.